src/nvim/decoration.c

```diff
@@ -847,8 +851,8 @@ static void buf_signcols_validate_range(buf_T *buf, int row1, int row2, int add)
       count++;
       if (mt_paired(mark)) {
         MTPos end = marktree_get_altpos(buf->b_marktree, mark, NULL);
-        for (int i = mark.pos.row; i < MIN(row2, end.row); i++) {
-          overlap[row2 - i]++;
+        for (int i = mark.pos.row + 1 - row1; i < MIN(row2, end.row); i++) {
```

Comment:
This loop range made no sense actually, was just the first thing I tried and didn't happen to fail any tests. Used my 🧠 to verify correctness and added a test to `decorations_spec.lua` that failed.